pkg/extensions/sync/sync.go

```diff
@@ -0,0 +1,436 @@
+
+	refs := *sourceReferences
+
+	for _, c := range content {
+		if c.Tags == nil {
+			continue
+		}
+
+		if c.Tags.Semver != nil {
+			for i := 0; i < len(refs); i++ {
+				tagged := getTagFromRef(refs[i], log)
+				if tagged == nil {
+					continue
+				}
+
+				_, ok := semver.NewVersion(tagged.Tag())
+				if ok != nil {
+					log.Info().Msgf("tag %s is not semver, skipping", tagged.Tag())
+
+					refs = append(refs[:i], refs[i+1:]...)
+					i--
+				}
+			}
+		}
+	}
+
+	*sourceReferences = refs
+}
+
+// imagesToCopyFromRepos lists all images given a registry name and its repos.
+func imagesToCopyFromRepos(registryName string, repos []string, sourceCtx *types.SystemContext,
+	log log.Logger) ([]types.ImageReference, error) {
+	var sourceReferences []types.ImageReference
+
+	for _, repoName := range repos {
+		repoRef, err := parseRepositoryReference(fmt.Sprintf("%s/%s", registryName, repoName))
+		if err != nil {
+			return nil, err
+		}
+
+		tags, err := getImageTags(context.Background(), sourceCtx, repoRef)
+		if err != nil {
+			log.Err(err).Msg("Error getting image tags")
+			return nil, err
+		}
+
+		for _, tag := range tags {
+			taggedRef, err := reference.WithTag(repoRef, tag)
+			if err != nil {
+				log.Err(err).Msgf("Error creating a reference for repository %s and tag %q", repoRef.Name(), tag)
+				return nil, err
+			}
+
+			ref, err := docker.NewReference(taggedRef)
+			if err != nil {
+				log.Err(err).Msgf("Cannot obtain a valid image reference for transport %q and reference %s",
+					docker.Transport.Name(), taggedRef.String())
+				return nil, err
+			}
+
+			sourceReferences = append(sourceReferences, ref)
+		}
+	}
+
+	return sourceReferences, nil
+}
+
+func syncRegistry(regCfg RegistryConfig, log log.Logger, destRegistry string, destCtx *types.SystemContext,
+	policyCtx *signature.PolicyContext) error {
+	locker.Lock()
+	defer locker.Unlock()
+
+	log.Info().Msgf("Syncing registry: %s", regCfg.URL)
+
+	srcCtx := &types.SystemContext{}
+	if regCfg.TLSVerify {
+		srcCtx.DockerCertPath = regCfg.CertDir
+		srcCtx.DockerDaemonCertPath = regCfg.CertDir
+		srcCtx.DockerDaemonInsecureSkipTLSVerify = false
+		srcCtx.DockerInsecureSkipTLSVerify = types.NewOptionalBool(false)
+	} else {
+		srcCtx.DockerDaemonInsecureSkipTLSVerify = true
+		srcCtx.DockerInsecureSkipTLSVerify = types.NewOptionalBool(true)
+	}
+
+	if regCfg.Credentials != (types.DockerAuthConfig{}) {
+		srcCtx.DockerAuthConfig = &regCfg.Credentials
+	}
+
+	options := copy.Options{
+		DestinationCtx: destCtx,
+		SourceCtx:      srcCtx,
+	}
+
+	catalog, err := getCatalog(regCfg)
+	if err != nil {
+		return err
+	}
+
+	registryName := strings.Replace(strings.Replace(regCfg.URL, "http://", "", 1), "https://", "", 1)
+
+	repos := filterRepos(catalog.Repositories, regCfg.Content)
+
+	images, err := imagesToCopyFromRepos(registryName, repos, srcCtx, log)
+	if err != nil {
+		return err
+	}
+
+	err = filterImagesByTagRegex(&images, regCfg.Content, log)
```

Comment:
Is it possible to call filterImagesByTagRegex and filterImagesBySemver inside imagesToCopyFromRepos https://github.com/peusebiu/zot/blob/sync/pkg/extensions/sync/sync.go#L292 ?

Reply:
Sure.